packages/core/src/data/strategies/SearchNativeFetchStrategy.ts

```diff
@@ -0,0 +1,155 @@
+import { getSiteBySourceUrl, addQueryArgs, getWPUrl } from '../../utils';
+import { endpoints } from '../utils';
+import { apiGet } from '../api';
+import { PostEntity, QueriedObject } from '../types';
+import { searchMatchers } from '../utils/matchers';
+import { parsePath } from '../utils/parsePath';
+import { FetchOptions, AbstractFetchStrategy, EndpointParams } from './AbstractFetchStrategy';
+
+/**
+ * The EndpointParams supported by the {@link SearchNativeFetchStrategy}
+ */
+export interface SearchParams extends EndpointParams {
+	/**
+	 * Current page of the collection.
+	 *
+	 * @default 1
+	 */
+	page?: number;
+
+	/**
+	 * Maximum number of items to be returned in result set.
+	 *
+	 * @default 10
+	 */
+	per_page?: number;
+
+	/**
+	 * Limit results to those matching a string.
+	 */
+	search?: string;
+
+	/**
+	 * Limit results to items of an object type.
+	 *
+	 * @default 'post'
+	 */
+	type?: 'post' | 'term' | 'post-format';
+
+	/**
+	 * Limit results to items of one or more object subtypes.
+	 */
+	subtype?: string;
+
+	/**
+	 * Ensure result set excludes specific IDs.
+	 */
+	exclude?: number[];
+
+	/**
+	 * Limit result set to specific IDs.
+	 */
+	include?: number[];
+}
+
+/**
+ * The SearchNativeFetchStrategy is used to fetch search results for a given search query
+ * Uses the native WordPress search endpoint.
+ *
+ * Note that custom post types and custom taxonomies should be defined in `headless.config.js`
+ *
+ * This strategy supports extracting endpoint params from url E.g:
+ * - `/page/2/` maps to `{ page: 2 }`
+ * - `/searched-term/page/2` maps to `{ search: 'searched-term', page: 2 }`
+ *
+ * @see {@link getParamsFromURL} to learn about url param mapping
+ *
+ * @category Data Fetching
+ */
+export class SearchNativeFetchStrategy<
+	T extends PostEntity = PostEntity,
```

Comment:
@nicholasio everything seems to work fine with these changes, but I am not sure if I need to create a new SearchPostEntity as the data is not the same structure as a post entity in other WP endpoints. But I am not quite understanding why it works without a new entity.

Reply:
Yeah the types will end up wrong and those using TS will notice it. We do need a new `SearchEntitty` the result can be either be a post or a term object.

Reply:
updated